### PR TITLE
added missing function db._transactions(), and equivalent REST API route GET /_api/transaction

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Added missing function `db._transactions()` plus REST route GET /_api/transaction

- [x] Bug-Fix for *devel-branch* (i.e. no need for backports?)
- [x] The behaviour change can only be verified via automatic tests

#### Related Information

- [x] There is an internal planning ticket: https://github.com/arangodb/release-3.5/issues/83

### Testing & Verification

This PR adds tests that were used to verify all changes:

- [x] Added new **integration tests** (i.e. in shell_server / shell_server_aql)

https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/5419/

### Documentation

- [x] Added a new section in the *http API* 
- [x] Added *Swagger examples* for the http API

### CODE REVIEW

File: Documentation/DocuBlocks/Rest/Transactions/get_api_transactions.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+@EXAMPLE_ARANGOSH_RUN{RestTransactionsGet}
+    db._drop("products");
+    db._create("products");
+    let body = {
+      collections: {
+        read : "products"
+      }
+    };
+    let trx = db._createTransaction(body);
```
Comment (graetzer) [inline Documentation/DocuBlocks/Rest/Transactions/get_api_transactions.md]: is this API public and visible ?

File: arangod/Transaction/Manager.cpp [hunk 2]
```diff
@@ -723,5 +746,71 @@ bool Manager::abortManagedTrx(std::function<bool(TransactionState const&)> cb) {
...
+          }
+          VPackSlice slice = builder.slice();
+          headers.emplace(StaticStrings::Authorization,
+                          "bearer " + auth->tokenCache().generateJwt(slice));
+        }
+      }
+  
+      double const timeout = 5.0;
+      
+      auto comres =
```
Comment (graetzer) [inline arangod/Transaction/Manager.cpp]: don't we want to get away from these kind of blocking APIs ? How about we revive the networking v2 PR ?

File: arangod/Transaction/Manager.h [hunk 3]
```diff
@@ -62,6 +67,30 @@ class Manager final {
 
  public:
   typedef std::function<void(TRI_voc_tid_t, TransactionData const*)> TrxCallback;
+  
+  enum class MetaType : uint8_t {
```
Comment (graetzer) [inline arangod/Transaction/Manager.h]: I think instead of making this public, how about just passing in the TransactionState* ?

**Comment by jsteemann** [comment]
https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/5430/

**Comment by jsteemann** [comment]
https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/5433/

**Comment by jsteemann** [comment]
https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/5436/

**Comment by jsteemann** [comment]
https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/5438/

**Comment by jsteemann** [comment]
Fixed MMFiles dead lock:
https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/5445/

**Pull request merged**